modules/core/src/test/scala/org/tessellation/domain/statechannel/StateChannelValidatorSuite.scala

```diff
@@ -167,23 +170,82 @@ object StateChannelValidatorSuite extends MutableIOSuite {
       )
   }
 
-  test("should succeed when the state channel seedlist check is disabled") { res =>
+  test("should fail when no signature is on the state channel whitelist for given address") { res =>
+    implicit val (kryo, sp) = res
+
+    for {
+      keyPair1 <- KeyPairGenerator.makeKeyPair[IO]
+      keyPair2 <- KeyPairGenerator.makeKeyPair[IO]
+      peerId1 = NonEmptySet.one(PeerId.fromPublic(keyPair1.getPublic))
+      peerId2 = NonEmptySet.one(PeerId.fromPublic(keyPair2.getPublic))
+      address = testStateChannel.toAddress
+      signedSCBinary <- forAsyncKryo(testStateChannel, keyPair2)
+      scOutput = StateChannelOutput(address, signedSCBinary)
+      validator = mkValidator(peerId1.union(peerId2).toSortedSet.toSet.some, Map(address -> peerId1).some)
+      result <- validator.validate(scOutput)
+    } yield
+      expect.same(
+        StateChannelValidator.NoSignerFromStateChannelWhitelist.invalidNec,
+        result
+      )
+  }
+
+  test("should succeed when at least one signature is on the state channel whitelist for given address") { res =>
+    implicit val (kryo, sp) = res
+
+    for {
+      keyPair1 <- KeyPairGenerator.makeKeyPair[IO]
+      keyPair2 <- KeyPairGenerator.makeKeyPair[IO]
+      peerId1 = NonEmptySet.one(PeerId.fromPublic(keyPair1.getPublic))
+      peerId2 = NonEmptySet.one(PeerId.fromPublic(keyPair2.getPublic))
+      address = testStateChannel.toAddress
+      signedSCBinary <- forAsyncKryo(testStateChannel, keyPair1).flatMap(_.signAlsoWith(keyPair2))
+      scOutput = StateChannelOutput(address, signedSCBinary)
+      validator = mkValidator(peerId1.union(peerId2).toSortedSet.toSet.some, Map(address -> peerId1).some)
+      result <- validator.validate(scOutput)
+    } yield expect.same(Valid(scOutput), result)
+  }
+
+  test("should succeed when the seedlist and state channel whitelist check is disabled") { res =>
     implicit val (kryo, sp) = res
 
     for {
       keyPair <- KeyPairGenerator.makeKeyPair[IO]
-      address = keyPair.getPublic.toAddress
+      address = testStateChannel.toAddress
       signedSCBinary <- forAsyncKryo(testStateChannel, keyPair)
       scOutput = StateChannelOutput(address, signedSCBinary)
-      validator = mkValidator(None)
+      validator = mkValidator(None, None)
       result <- validator.validate(scOutput)
     } yield expect.same(Valid(scOutput), result)
   }
 
-  private def mkValidator(stateChannelSeedlist: Option[Set[Address]], maxBinarySizeInBytes: Long = 1024)(
+  test("should fail when address is not gerated from data for genesis state channel") { res =>
```

Comment:
`gerated`?|
maybe:
`should fail when address is not derived from genesis of the state channel`?

Reply:
Rephrased.